espree.js

```diff
@@ -3857,6 +3858,15 @@ function parseParams(firstRestricted) {
                     message = Messages.StrictParamDupe;
                 }
             }
+            if (match("=")) {
+                if (!allowDefaultParams) {
+                    throwUnexpected(lookahead);
+                }
+                lex();
+                def = parseAssignmentExpression();
+                ++defaultCount;
+            }
+            defaults.push(def);
```

Comment:
Gotcha, that makes sense. Then is the `defaultCount === 0` check below still necessary?